File: Leetcode/linkedlist/203_deleteNodeFromLinkedList/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        //if (head!=null&& head.next==null)
            //if (head.val == val )
            //    return null;
          //  else
        //        return head;
        ListNode temp = head;
        if (temp == null)
            return null;
        while(temp.next!=null){
            if (temp.val!=val && temp.next.val != val)
                temp = temp.next;
            else if(temp.val == val){
                temp.val = temp.next.val;
                temp.next = temp.next.next;
            }
            else if(temp.next.val == val){
                temp.next = temp.next.next;
            }
        }
        if (temp.val == val && temp.next ==null)
            return null;
        return head;
    }
}

File: Leetcode/linkedlist/203_deleteNodeFromLinkedList/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        dummyHead = ListNode(-1)
        temp = dummyHead
        dummyHead.next = head
        while temp and temp.next:
            while temp.next and temp.next.val==val:
                temp.next = temp.next.next
            temp = temp.next
        return dummyHead.next
